src/classes/InventoryApis/InventoryApi.ts

```diff
@@ -0,0 +1,180 @@
+import axios, { AxiosError } from 'axios';
+import filterAxiosError from '@tf2autobot/filter-axios-error';
+import { UnknownDictionary, UnknownDictionaryKnownValues } from 'src/types/common';
+import SteamID from 'steamid';
+import CEconItem from '@tf2autobot/steamcommunity/classes/CEconItem';
+import { EconItem } from '@tf2autobot/tradeoffer-manager';
+
+import Bot from '../Bot';
+
+export default class InventoryApi {
+    constructor(private bot: Bot, private optionsKey: string) {}
+
+    public isEnabled(): boolean {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-member-access
+        return this.bot.options.inventoryApis[this.optionsKey].enable;
+    }
+
+    public getApiKey(): string {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+        return this.bot.options[this.optionsKey + 'ApiKey'];
+    }
+
+    // This method should be defined by a class extending InventoryAPI
+    protected getURLAndParams(
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        steamID64: string,
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        appID: number,
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        contextID: string
+    ): [string, UnknownDictionaryKnownValues] {
+        return ['', {}];
+    }
+
+    // Adapted from node-steamcommunity
+    public getUserInventoryContents(
+        userID: SteamID | string,
+        appID: number,
+        contextID: string,
+        tradableOnly: boolean,
+        callback: (err?: Error, inventory?: EconItem[], currency?: EconItem[], totalInventoryCount?: number) => void
+    ): void {
+        if (!this.isEnabled()) {
+            callback(new Error("This API is disabled in the bot's options"));
```

Comment:
I wonder whether this error message will be shown to whoever that's communicating with the bot or not.
If not, then it's alright I guess.

Reply:
`InventoryGetter` checks whether each inventory API is enabled before using this method on it, so this error should not ever come up in normal bot usage.

Reply:
I see. Alright thank you very much.